src/main/java/ai/aliz/talendtestrunner/service/TestRunnerService.java

```diff
@@ -39,10 +39,29 @@
     private final AssertActionService assertActionService;
     private final TalendJobStateChecker talendJobStateChecker;
     private final BigQueryExecutor bigQueryExecutor;
+    private final ExecutionActionService executionActionService;
     
     public void runTest(ai.aliz.talendtestrunner.testconfig.TestCase testCase) {
         initActionService.run(testCase.getInitActionConfigs(), contextLoader);
-//        testCase.getExecuteActions().forEach(executeAction -> runTalendJob(contextLoader, executeAction, testCase));
+
+        testCase.getExecutionActionConfigs().forEach(executionActionConfig -> {
+            switch (executionActionConfig.getType()) {
+                case AirFlow:
+                    break;
+                case BqQuery:
+                    executeBQQuery(TestRunnerUtil.getSourceContentFromConfigProperties(executionActionConfig), contextLoader.getContext("local"));
```

Comment:
you need use here 2 context. The local context is needed to read the query file, and a BQ context is needed to replace the placeholders in the query